src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -325,10 +373,90 @@ internal static bool ValidateParentTagName(string parentTag, ErrorSink errorSink
                     errorSink: errorSink);
         }
 
-        private static bool ValidateName(
-            string name,
-            bool targetingAttributes,
-            ErrorSink errorSink)
+        private static bool ValidateRequiredAttribute(string attribute, ErrorSink errorSink)
```

Comment:
I tried that approach. It turned out to be a huge re-working of how the `TagHelperDescriptorFactory` currently works. Going to stick with the duplication.